pkg/virt-launcher/virtwrap/sriov/hostdev.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package sriov
+
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/pkg/util"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+func CreateHostDevices(vmi *v1.VirtualMachineInstance) ([]api.HostDevice, error) {
+	var hostDevices []api.HostDevice
+
+	pciAddrPool := NewPCIAddressPool(vmi.Spec.Domain.Devices.Interfaces)
+
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if iface.SRIOV == nil {
+			continue
+		}
```

Comment:
We are already doing this check on the pool before inserting elements, maybe we need some kind of "Iterator" in the pool so we do 

```golang
for _, iface := range pciAddrPool.Ifaces() {
}
```

Reply:
I don't see the ownership problem, at the end we are iterating all the pool and creating all the host devices from there.

Reply:
I am ok with the `sriovInterfaces(vmi)` so I supose we will have something like.

```golang
sriovInterfaces = filterSriovInterfaces(vmi)
pciAddrPool := NewPCIAddressPool(sriovIntefaces)

	for _, iface := range sriovInterfaces {
```

Reply:
Done